resources/task.rb

```diff
@@ -0,0 +1,149 @@
+property :task_name, String, desired_state: false, identity: true, name_attribute: true
+property :task_source, String, desired_state: false, identity: true, default: ''.freeze
+property :task_crontab, String, desired_state: false, identity: true, default: '0 1 * * * ?'.freeze
+property :api_endpoint, String, desired_state: false, identity: true,
+                                default: lazy { node['nexus3']['api']['endpoint'] }
+property :api_username, String, desired_state: false, identity: true,
+                                default: lazy { node['nexus3']['api']['username'] }
+property :api_password, String, desired_state: false, identity: true, sensitive: true,
+                                default: lazy { node['nexus3']['api']['password'] }
+
+load_current_value do |desired|
+  api_endpoint desired.api_endpoint
+  api_username desired.api_username
+  api_password desired.api_password
+
+  def apiclient
+    @apiclient ||= ::Nexus3::Api.new(api_endpoint, api_username, api_password)
+  end
```

Comment:
instead of doing a local function, please use a variable

Reply:
done